src/visitor/DeducePropsVisitor.h

```diff
@@ -18,7 +18,67 @@ class Expression;
 namespace graph {
 
 class QueryContext;
-class ExpressionProps;
+
+class ExpressionProps final {
```

Comment:
I prefer to move the ExpressionProps to visitor directory rather than merge it with visitor into single file.

Reply:
It's related to this visitor only.